src/components/Header/Header.js

```diff
@@ -125,26 +122,20 @@ class Header extends React.Component {
   }
 
   componentWillUnmount() {
-    HeaderStore.unlisten(this.onChange.bind(this));
     // Listen on FeatureFlags Store updates
     FeatureFlags.store.unlisten(this.onFeatureFlagsChange.bind(this));
     // Removing event listener to minimize garbage collection
   }
 
   onChange() {
```

Comment:
Is this a core react hook? Don't see it documented outside of html5 interactive elements. Is it fired when the component change?

Reply:
It is a core react hook - https://reactjs.org/docs/react-component.html#componentwillunmount

It gets fired when a component gets removed from the DOM.

Reply:
Sorry I meant `onChange`

Reply:
Ah, it's not. That function can be named anything, but `onChange` is a convention. It's the function that's used to listen and not listen to the Store. http://alt.js.org/guide/view/

Reply:
I see! Thanks. Looks like, with the removal of the store, `onChange` is no longer called, so the whole method can be deleted https://github.com/NYPL/dgx-header-component/blob/44df90663cb0df9d0ce5e5cfbe18bf7cdc72f98c/src/components/Header/Header.js#L130-L138

Reply:
Correct, I did delete it just recently.